src/citrine/resources/dataset.py

```diff
@@ -273,6 +276,18 @@ def delete_contents(
                                   dataset_uid=self.uid,
                                   project_id=self.project_id
                                   )
+        if prompt_to_confirm:
+            print(f"Do you really want to delete the contents of dataset {self.uid}? [Y/N]")
+            user_response = None
+            while user_response not in {'Y', 'N'}:
+                user_response = input()
+            if user_response == 'Y':
+                pass
+            elif user_response == 'N':
+                print(f'Aborting deletion of dataset {self.uid} contents')
+                return []
+            else:
+                raise ValueError(f'Expected user response of Y or N. Got {user_response}')
```

Comment:
```suggestion
        while(prompt_to_confirm):
            print(f"Confirm you want to delete the contents of {self.uid} [Y/N]")
            user_response = input()
            if user_response.lower() in {'y', 'yes'}:
                break  # return to main flow
            elif user_response.lower() in {'n', 'no'}:
                raise RuntimeError("delete_contents was invoked unintentionally")
            else:
                print(f'"{user_response}" is not a valid response')
```

I could see `return []` as possibly more appropriate that the exception, but the rational in PLA-8079 was accidental execution of a Jupyter cell -- in that scenario it seems more correct to stop execution and allow for manual intervention rather than trying to execute the next thing in the cell (or the next cell).  This user verification step should not be relied upon for flow control at the user level.

Reply:
I'd used `break` instead of `prompt_to_confirm = False` (which would have the same functional effect) because I thought there might be value in knowing what the passed parameter was, or at least there was no benefit in clobbering that value.